client/discovery/discovery.js

```diff
@@ -0,0 +1,187 @@
+window.onload = function() {
+  if(typeof switch_is_present == 'undefined') mountDiscovery()
```

Comment:
remember your truthy and falsy attributes, if you are checking for undefined, couldn't you just do if (!switch_is_present) ?